File: fix_layout.py
import re

def fix_file(filepath):
    with open(filepath, "r", encoding="utf-8") as f:
        content = f.read()

    # 1. Fix balanceRecovery missing admin-card
    # Find balanceRecovery and fix the div inside it
    if 'id="balanceRecovery"' in content:
        # We know it has <div style="margin-bottom:28px;"> right before <div class="card-header">
        # Let's replace it carefully inside the balanceRecovery section
        pattern = r'(id="balanceRecovery"[\s\S]*?)(<div style="margin-bottom:28px;">)(\s*<div class="card-header">)'
        content = re.sub(pattern, r'\1<div class="admin-card">\3', content)

    # 2. Rewrite tshirtSection to be cleanly wrapped and without Coordinators
    # We will extract the application form and sizes overview
    # and put them into proper .admin-card wrappers.
    # First, let's find the Application Form HTML block
    app_form_pattern = r'(<!-- Application Form \+ Admin Price Settings -->.*?)(?=<!-- Sizes Overview -->)'
    app_form_match = re.search(app_form_pattern, content, re.DOTALL)
    if not app_form_match:
        print(f"Could not find Application Form in {filepath}")
        return

    app_form_html = app_form_match.group(1).strip()
    # Let's remove the inline style wrapper for the form to let admin-card handle it
    # We will just keep it as is since it has its own grid and backgrounds, but we'll put it in an admin-card or just leave it since the grid items look like cards already.
    # Wait, the app form uses: <div style="display:grid..."><div style="background:#fff;border-radius:12px;...">
    # Those inner divs ARE essentially cards.
    # We can just wrap the whole tshirtSection properly.

    sizes_pattern = r'(<!-- Sizes Overview -->.*?)(?=</div>\s*</div>\s*</div>\s*<!-- \S* END)'
    # Wait, the end of tshirtSection is a bit messy to regex. Let's just grab the whole tshirtSection
    tshirt_pattern = r'(<!-- ---------- T-SHIRT SECTION ---------- -->\s*<div id="tshirtSection".*?>).*?(</div>\s*</div>\s*</div>)'
    tshirt_match = re.search(tshirt_pattern, content, re.DOTALL)
    if not tshirt_match:
        print(f"Could not find full tshirtSection in {filepath}")
        # fallback for dashboard.html which might not have the comment exact match
        tshirt_pattern2 = r'(<div id="tshirtSection" class="content-section".*?>).*?(</div>\s*</div>\s*</div>)'
        tshirt_match = re.search(tshirt_pattern2, content, re.DOTALL)
        if not tshirt_match:
            print(f"Still could not find tshirtSection in {filepath}")
            return

    # Now let's extract just the parts we want:
    # 1. The forms grid
    forms_match = re.search(r'(<div style="display:grid;grid-template-columns:repeat.*?</div>\s*</div>\s*</div>)', tshirt_match.group(0), re.DOTALL)
    forms_html = forms_match.group(1) if forms_match else ""

    # 2. Sizes Overview
    sizes_match = re.search(r'(<!-- Sizes Overview -->.*?</div>\s*</div>)', tshirt_match.group(0), re.DOTALL)
    sizes_html = sizes_match.group(1) if sizes_match else ""

    display_style = ' style="display:none;"' if 'dashboard.html' in filepath else ''

    new_tshirt_section = f"""<!-- ---------- T-SHIRT SECTION ---------- -->
                <div id="tshirtSection" class="content-section"{display_style}>
                    <div class="page-title">
                        <h1><i class="fas fa-tshirt" style="color:var(--primary-color);margin-right:10px;"></i>T-shirt Section</h1>
                        <p>Manage T-shirt applications and size summaries</p>
                    </div>

                    <!-- Application Form + Admin Price Settings -->
                    {forms_html}

                    {sizes_html}
                </div>"""

    # Replace the old tshirtSection with the new one
    # We replace from <div id="tshirtSection"... up to the last closing div of the section
    content = content[:tshirt_match.start()] + new_tshirt_section + content[tshirt_match.end():]

    with open(filepath, "w", encoding="utf-8") as f:
        f.write(content)
    print(f"Fixed layout in {filepath}")

fix_file("dashboard.html")
fix_file("admin.html")

File: fix_layout.js
const fs = require("fs");

function fixFile(filepath) {
    let content = fs.readFileSync(filepath, "utf-8");

    // 1. Fix balanceRecovery missing admin-card
    if (content.includes('id="balanceRecovery"')) {
        const pattern = /(id="balanceRecovery"[\s\S]*?)(<div style="margin-bottom:28px;">)(\s*<div class="card-header">)/;
        content = content.replace(pattern, '$1<div class="admin-card">$3');
    }

    // 2. Rewrite tshirtSection
    const formMatch = content.match(/<!-- Application Form \+ Admin Price Settings -->[\s\S]*?(?=<!-- Sizes Overview -->)/);
    if (!formMatch) return console.log("Form not found in", filepath);
    
    // We want to grab sizesMatch up to the <p>Click any size...
    const sizesMatch = content.match(/<!-- Sizes Overview -->[\s\S]*?<p style="margin:12px 0 0;font-size:.78rem;color:#aaa;text-align:center;">Click any size to view applicants<\/p>\s*<\/div>\s*<\/div>/);
    if (!sizesMatch) return console.log("Sizes not found in", filepath);

    let displayStyle = filepath.includes("dashboard") ? ' style="display:none;"' : '';

    const newTshirtSection = `<!-- ══════════ T-SHIRT SECTION ══════════ -->
                <div id="tshirtSection" class="content-section"${displayStyle}>
                    <div class="page-title">
                        <h1><i class="fas fa-tshirt" style="color:var(--primary-color);margin-right:10px;"></i>T-shirt Section</h1>
                        <p>Manage T-shirt applications and size summaries</p>
                    </div>

                    <div class="admin-card">
                        ${formMatch[0].trim()}
                        
                        <div style="margin-top:28px;">
                            ${sizesMatch[0].trim()}
                        </div>
                    </div>
                </div>\n`;

    // Now find where to replace
    const startIdx = content.indexOf('<div id="tshirtSection"');
    if (startIdx === -1) return console.log("Start not found");
    
    const sizesEndText = '<p style="margin:12px 0 0;font-size:.78rem;color:#aaa;text-align:center;">Click any size to view applicants</p>\n        </div>\n    </div>\n</div>';
    let endIdx = content.indexOf(sizesEndText, startIdx);
    
    if (endIdx === -1) {
        // fallback
        const altEnd = '<p style="margin:12px 0 0;font-size:.78rem;color:#aaa;text-align:center;">Click any size to view applicants</p>';
        endIdx = content.indexOf(altEnd, startIdx);
        if (endIdx === -1) return console.log("End not found in", filepath);
        
        // Find the 3 closing divs after it
        let temp = content.substring(endIdx);
        const match = temp.match(/<\/p>[\s]*<\/div>[\s]*<\/div>[\s]*<\/div>/);
        if (match) {
            endIdx = endIdx + match.index + match[0].length;
        } else {
            endIdx = endIdx + altEnd.length;
        }
    } else {
        endIdx = endIdx + sizesEndText.length;
    }

    let replaceStart = startIdx;
    const commentStr = '<!-- ══════════ T-SHIRT SECTION ══════════ -->\n                ';
    // check if comment exists
    const beforeStart = content.substring(Math.max(0, startIdx - 100), startIdx);
    if (beforeStart.includes('<!-- ══════════ T-SHIRT SECTION ══════════ -->')) {
        replaceStart = content.lastIndexOf('<!-- ══════════ T-SHIRT SECTION ══════════ -->', startIdx);
    } else if (beforeStart.includes('<!-- \ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd\ufffd T-SHIRT SECTION')) {
        replaceStart = content.lastIndexOf('<!--', startIdx);
    } else if (beforeStart.includes('<!--  ? ? ? ? ? ? ? ? ? ? T-SHIRT SECTION')) {
        replaceStart = content.lastIndexOf('<!--', startIdx);
    }

    content = content.substring(0, replaceStart) + newTshirtSection + content.substring(endIdx);
    
    fs.writeFileSync(filepath, content);
    console.log("Fixed", filepath);
}

fixFile("dashboard.html");
fixFile("admin.html");
